vllm/spec_decode/batch_expansion.py

```diff
@@ -159,11 +160,18 @@ def _contract_batch(
         target_sampler_output will be contracted to.
         """
         contracted_bs = len(contracted_seq_group_metadata_list)
-        (target_token_ids, target_probs, target_logprobs, target_hidden_states,
-         non_spec_target_token_ids, non_spec_target_probs,
-         non_spec_target_logprobs,
-         non_spec_target_hidden_states) = self._split_scoring_output(
-             target_sampler_output, num_scoring_tokens)
+        if current_platform.is_hpu():
+            (target_token_ids, target_probs, target_logprobs,
+             target_hidden_states, non_spec_target_token_ids,
+             non_spec_target_probs, non_spec_target_logprobs,
+             non_spec_target_hidden_states) = self._split_scoring_output_hpu(
+                 target_sampler_output, num_scoring_tokens)
+        else:
+            (target_token_ids, target_probs, target_logprobs,
+             target_hidden_states, non_spec_target_token_ids,
+             non_spec_target_probs, non_spec_target_logprobs,
+             non_spec_target_hidden_states) = self._split_scoring_output(
+                 target_sampler_output, num_scoring_tokens)
```

Comment:
Output and invocation is a duplication of existing, just different method is called, with the _hpu suffix. This can be simplified by calling the _hpu method in the _split_scoring_output based on is_hpu() value.

Reply:
The reason I want to change in caller not in the function is because I found 'self._split_scoring_output' is updated consistently and I want to avoid potential conflict in future, Adding a new self._split_scoring_output_hpu() will be easier to maintain.